systems/snl-eldorado/system.py

```diff
@@ -0,0 +1,730 @@
+                },
+                "texinfo": {"externals": [{"spec": "texinfo@6.5", "prefix": "/usr"}]},
+                "bison": {"externals": [{"spec": "bison@3.0.4", "prefix": "/usr"}]},
+                "python": {
+                    "externals": [
+                        {
+                            "spec": "python@3.9.12",
+                            "prefix": "/usr/tce/packages/python/python-3.9.12",
+                            "buildable": False,
+                        }
+                    ]
+                },
+                "unzip": {
+                    "buildable": False,
+                    "externals": [{"spec": "unzip@6.0", "prefix": "/usr"}],
+                },
+                "hypre": {"variants": f"amdgpu_target={self.rocm_arch}"},
+                "hwloc": {
+                    "externals": [
+                        {"spec": "hwloc@2.9.1", "prefix": "/usr", "buildable": False}
+                    ]
+                },
+                "fftw": {"buildable": False},
+                "intel-oneapi-mkl": {
+                    "externals": [
+                        {
+                            "spec": "intel-oneapi-mkl@2023.2.0",
+                            "prefix": "/opt/intel/oneapi",
+                        }
+                    ],
+                    "buildable": False,
+                },
+                "fftw-api": {
+                    "buildable": False,
+                    "require": "intel-oneapi-mkl",
+                },
+                "mpi": {"buildable": False},
+                "libfabric": {
+                    "externals": [
+                        {"spec": "libfabric@2.1", "prefix": "/opt/cray/libfabric/2.1"}
+                    ],
+                    "buildable": False,
+                },
+            }
+        }
+
+        selections["packages"] |= self.rocm_config()["packages"]
+
+        selections["packages"] |= self.mpi_config()["packages"]
+
+        if self.spec.satisfies("compiler=cce"):
+            selections["packages"] |= {
+                "cray-libsci": {
+                    "externals": [
+                        {
+                            "spec": "cray-libsci@23.05.1.4%cce",
+                            "prefix": "/opt/cray/pe/libsci/23.05.1.4/cray/12.0/x86_64/",
+                        }
+                    ]
+                }
+            }
+        elif self.spec.satisfies("compiler=gcc"):
+            selections["packages"] |= {
+                "cray-libsci": {
+                    "externals": [
+                        {
+                            "spec": "cray-libsci@23.05.1.4%gcc",
+                            "prefix": "/opt/cray/pe/libsci/23.05.1.4/gnu/10.3/x86_64/",
+                        }
+                    ]
+                }
+            }
+
+        selections["packages"] |= self.compiler_weighting_cfg()["packages"]
+
+        return selections
+
+    def compiler_weighting_cfg(self):
```

Comment:
I think you can entirely replace this with the implementation of `compiler_weighting_cfg` from https://github.com/LLNL/benchpark/blob/develop/systems/llnl-elcapitan/system.py